PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ConnectionStringCheck.java

```diff
@@ -0,0 +1,44 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.*;
+import com.intellij.psi.*;
+import com.microsoft.applicationinsights.TelemetryClient;
+import org.jetbrains.annotations.NotNull;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class extends the LocalInspectionTool and checks for the usage of connection strings in the code.
+ * Connection strings that start with "DefaultEndpointsProtocol=http;AccountName=" are flagged as problems.
+ * This is because the use of such connection strings is not recommended.
+ */
+public class ConnectionStringCheck extends LocalInspectionTool {
+
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+
+        TelemetryClient telemetry = TelemetryClientProvider.getTelemetryClient();
+
+        return new JavaElementVisitor() {
+
+            @Override
+            public void visitLiteralExpression(PsiLiteralExpression expression) {
+                super.visitLiteralExpression(expression);
+
+                String value = expression.getValue() instanceof String ? (String) expression.getValue() : null;
+
+                if (value != null && value.startsWith("DefaultEndpointsProtocol=http;AccountName=")) {
+                    holder.registerProblem(expression, "Use of connection strings is not recommended");
```

Comment:
Will change to something like "Storing connection strings directly in tracked files is discouraged" in the next commit.